sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -132,17 +135,49 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
         if (promises.length >= maxPromisesAllowed) {
           // Waits for all be processed.
           const insertResults = await Promise.all(promises);
+          await this.saveBulkOfferingParentId(
+            insertResults.map((result) => result.createdOfferingId),
+            offeringRepo,
+          );
```

Comment:
I'm trying to understand this part of the code. Can you briefly explain it?  I feel like this part is a little redundant if we can push `insertResults` to `saveBulkOfferingParentId` outside the loop from `allResults` before `allResults` push `finalResults`. I could be wrong about this. Let me know your thoughts.

Reply:
@lewischen-aot `allResults` can have all the offerings in the file, which can exceed the IN() limit. That is the reason we are doing the update as a batch.

Reply:
Thanks @ann-aot I see that now. Conversation is resolved.